src/test/kotlin/org/malachite/estella/queue/TaskResultTest.kt

```diff
@@ -0,0 +1,138 @@
+    @Autowired
+    lateinit var rabbitTemplate: RabbitTemplate
+
+    @Autowired
+    private lateinit var taskRepository: TaskRepository
+
+    @Autowired
+    private lateinit var jobSeekerRepository: JobSeekerRepository
+
+    @Autowired
+    private lateinit var recruitmentStageRepository: RecruitmentStageRepository
+
+    @Autowired
+    private lateinit var applicationRepository: ApplicationRepository
+
+    @Autowired
+    private lateinit var applicationStageDataRepository: ApplicationStageRepository
+
+    @Autowired
+    private lateinit var taskStageRepository: TaskStageRepository
+
+
+    private lateinit var task: Task
+    private lateinit var taskStage: TaskStage
+
+    @BeforeEach
+    fun prepareTask() {
+        task = Task(null, SerialBlob("xd".toByteArray()), SerialClob("xd".toCharArray()), "filename", 10)
+        val jobseeker = jobSeekerRepository.findAll().first()
+
+        val stage = recruitmentStageRepository.findAll().first()
+        val application = Application(null,
+                Date(Calendar.getInstance().time.time),
+                ApplicationStatus.IN_PROGRESS,
+                jobseeker,
+                mutableSetOf(),
+                mutableListOf())
+        val savedApplication = applicationRepository.save(application)
+        val applicationStageData = ApplicationStageData(null, stage, savedApplication, null, null)
+        val savedApplicationStageData = applicationStageDataRepository.save(applicationStageData)
+        this.taskStage = TaskStage(null, listOf(), savedApplicationStageData)
+        taskStage = taskStageRepository.save(taskStage)
+        task = taskRepository.save(task)
+    }
+
+    @Test
+    fun `task result consuming`() {
+        expectThat(taskStage.tasksResult.size).isEqualTo(0)
+
+        val taskResult = TaskResult(null, SerialBlob("xd".toByteArray()), SerialClob("xd".toCharArray()), null, null, task, taskStage)
+        publish(taskResult)
+
+        eventually {
+            taskStage = taskStageRepository.findById(taskStage.id!!).get()
+            expectThat(taskStage.tasksResult.size).isEqualTo(1)
+        }
+    }
+
+    @Test
+    fun `task bad message format consuming`() {
+        expectThat(taskStage.tasksResult.size).isEqualTo(0)
+
+        badPublish()
+
+        coolDown {
+            expectThat(taskStage.tasksResult.size).isEqualTo(0)
+        }
+    }
+
+
+    fun publish(result: TaskResult) {
+        try {
+            val resultBody = mapOf(
+                    "results" to String(result.results.binaryStream.readAllBytes()),
+                    "code" to result.code.characterStream.readText(),
+                    "startTime" to result.startTime.toString(),
+                    "endTime" to result.endTime.toString(),
+                    "solverId" to result.taskStage.id!!.toString(),
+                    "taskId" to result.task.id!!.toString()
+            )
+            val msg = Json.encodeToString(resultBody)
```

Comment:
to Ci serializuje do Base64, że jesteś w stanie to potem zdekodować Base64?

Reply:
nie, ale test i tak przechodził więc zauważyłem dopiero dzisiaj xD